submitter/relayer/relayer.go

```diff
@@ -272,6 +295,20 @@ func (rl *Relayer) resendSecondTxOfCheckpointToBTC(tx2 *types.BtcTxInfo, bumpedF
 	return tx2, nil
 }
 
+func (rl *Relayer) inMempool() (bool, error) {
+	_, err := rl.GetRawTransaction(rl.lastSubmittedCheckpoint.Tx2.TxId)
+	if err != nil {
+		if strings.Contains(err.Error(), TxNotFoundErr.Error()) {
+			return false, nil
+		}
+		// Return the error if it's not a "not found" error
+		return false, err
+	}
+
+	// the transaction is in the mempool
```

Comment:
here indicates that either the tx is either in the mempool or on the chain. I think we only need to resend tx2 if it is in the mempool or it can't be found at all. Can we differentiate it?